openedx/frontend-app-course-authoring
# Upgrade frontend-build to v14

## Description

Upgrades dependencies for frontend-build, frontend-package.

Package | Change | Type |
-- | -- | -- |
@edx/frontend-platform | 7.0.1 -> 8.0.3 | major |
@edx/frontend-component-footer | 13.0.2 -> 14.0.3 | major |
@edx/react-unit-test-utils | 2.0.0 -> 2.1.1 | minor |
@edx/frontend-component-header | 5.1.0 -> 5.3.3 | minor |
@openedx/frontend-plugin-framework | 1.1.0 -> 1.2.1 | minor |
@openedx/paragon | 22.2.1 -> 22.5.1 | minor |
classnames | 2.2.6 -> 2.5.1 | minor |
core-js | 3.8.1 -> 3.37.1 | minor |
formik | 2.2.6 -> 2.4.6 | minor |
moment | 2.29.4 -> 2.30.1 | minor |
react-router | 6.16.0 -> 6.23.1 | minor |
react-router-dom | 6.16.0 -> 6.23.1 | minor |
react-textarea-autosize | 8.4.1 > 8.5.3 | minor |
@edx/brand | 1.2.2 -> 1.2.3 | patch |
@edx/frontend-component-ai-translations | 2.0.0 -> TBD, [PR](https://github.com/openedx/frontend-component-ai-translations/pull/16) | patch |
@edx/frontend-lib-content-components | 2.1.9 -> TBD, [PR](https://github.com/openedx/frontend-lib-content-components/pull/475) | patch |


## Supporting information

https://github.com/openedx/frontend-app-course-authoring/issues/1034

Private-ref: [FAL-3750](https://tasks.opencraft.com/browse/FAL-3750)

### Dependencies

Blocked by:

* https://github.com/openedx/frontend-lib-content-components/pull/475
* https://github.com/openedx/frontend-component-ai-translations/pull/16

Based on these PRs, which are likely to merge quickly:

* https://github.com/openedx/frontend-app-course-authoring/pull/1061
* https://github.com/openedx/frontend-app-course-authoring/pull/1062

## Testing instructions

CI should cover most of the functionality, but manual testing is needed to ensure essential features are not broken.

1. Enable all the [`new_studio` toggles](https://github.com/openedx/edx-platform/blob/master/cms/djangoapps/contentstore/toggles.py).
2. Visit as many pages in the Course Authoring MFE as you can, and check for errors.
